viewing_party/party.py

```diff
@@ -1,23 +1,177 @@
 
+def get_watched_avg_rating(user_data):
+    avg_rating = 0
+    for i in range(len(user_data["watched"])):
+        avg_rating += user_data["watched"][i]["rating"]
+    if len(user_data["watched"]) == 0:
+        return avg_rating
+    avg_rating /= len(user_data["watched"])
+    return avg_rating
+
+#second swing at get_watched_avg_rating. Which one is better?
+# def get_watched_avg_rating(user_data):
+#     user_watched = user_data["watched"]
+#     avg_rating = 0
+    
+#     for movie in user_watched:
+#         avg_rating += movie["rating"]
+#         print(avg_rating)
+#     if len(user_watched) == 0:
+#         return 0.0
+#     else:
+#         avg_rating /= len(user_watched)
+#         return avg_rating
+
+def get_most_watched_genre(user_data):
+    genre_dict = {}
+    i = 0
+    if len(user_data["watched"]) == 0:
+        return None
+    for key, value in user_data["watched"][i].items():
+        if key == "genre":
+            if value in genre_dict:
+                genre_dict[value] += 1
+            else:
+                genre_dict[value] = 1
+        i += 1
+    max_value = max(genre_dict.values())
+    for key, value in genre_dict.items():
+        if value == max_value:
+            return key
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
-        
+#returns unique movies that user has watched, but friends have not
+def get_unique_watched(user_data):
+    unique_watched_list = []
+
+    friend_watched_set = set(get_friends_watched(user_data))
+    user_watched_set = set(get_users_watched(user_data))
+
+    unique_set = user_watched_set - friend_watched_set
+
+    for movie in unique_set:
+        for i in range(len(user_data["watched"])):
+            if user_data["watched"][i]['title'] == movie:
+                unique_watched_list.append(user_data["watched"][i])   
+
+    return unique_watched_list
+    
+    
+    
+def get_friends_watched(user_data):   
+    friends_list_of_dict = user_data["friends"]
+    friends_watched_list = []
+
+    for i in range(len(friends_list_of_dict)):
+        for watched in friends_list_of_dict[i]:
+            for movie in friends_list_of_dict[i][watched]:
+                friends_watched_list.append(movie['title'])
+
+    return friends_watched_list
+
+def get_users_watched(user_data):
+    user_watched_list = user_data["watched"]
+    users_watched_movies = []
+
+    for movie in user_watched_list:
+        users_watched_movies.append(movie['title'])
+
+    return users_watched_movies
+
+#returns what friends have watched, but user has not
+def get_friends_unique_watched(user_data):
+    unique_watched_list = []
+
+    friend_watched_set = set(get_friends_watched(user_data))
+    user_watched_set = set(get_users_watched(user_data))
+
+    unique_set = friend_watched_set - user_watched_set
+    
+    for i in range(len(user_data["friends"])):
+        for watched in user_data["friends"][i]:

```

Comment:
This loop is the same bug as in `get_friends_watched`.
```suggestion
        watched = user_data["friends"][i]["watched"]

```